### PR TITLE
feat(test): Adding betelgeuse docstrings to test_client.py

### PR DESCRIPTION
**Pull request by zpetrace**
Adding betelgeuse docstrings to test_client.py

### CODE REVIEW

**Review by Lorquas — COMMENTED** [review]
Looks good to me.  How are you generating the test cases and/or adding them to the junit.xml with betelgeuse? I would like to test this out directly...

**Comment by ptoscano** [comment]
> I would like to test this out directly...

Indeed -- this definitely needs to be tested with a github workflow/job, to verify that the format of these various comments, otherwise IMHO it's too easy to screw them up accidentally.

**Comment by Lorquas** [comment]
Other note, don't we need an `:id: <uuid>` field?

**Comment by Lorquas** [comment]
Update from slack convo: 

- We're going to co-opt this PR and add a README for the various manual betelgeuse invocations
- This PR will be blocked by CCT-675 where we will attempt to add .github/workflow and other automations (using this branch as a starting point)

File: integration-tests/test_client.py [hunk 1]
```diff
@@ -78,15 +122,24 @@ def test_client_logdir_permissions():
...
+        4. Verify that 2 new log files were created
+        5. Verify the size of insights-client.log
+        6. Verify the size of insights-client-payload.log
+    :expectedresults:
+        1. The logrotate config file exists
+        2. The insights-client registers successfully and logs are populated
+        3. The logrotate command is executed successfully
+        4. Two new log files were created
+        5. The size is 0B
+        6. The size is 0B
```
Comment (ptoscano) [inline integration-tests/test_client.py:142]: "the size" ... of what?

File: integration-tests/test_client.py [hunk 2]
```diff
@@ -131,8 +184,20 @@ def test_verify_logrotate_feature(insights_client):
...
+    :title: Verify insights-client details file exists
+    :description: Verify that the file /var/lib/insights/insights-client.json exists
+    :tier: Tier 0
+    :steps:
+        1. Register the insights-client
+        2. Delete /var/lib/insights/insights-client.json if it exists
+        3. run the --check-results command
+        4. Verify the existence of /var/lib/insights/insights-client.json
+    :expectedresults:
+        1. Insights-client is registrated
```
Comment (ptoscano) [inline integration-tests/test_client.py:197]: typo "registrated"

File: integration-tests/test_client.py [hunk 3]
```diff
@@ -55,9 +84,18 @@ def test_client_rpm_mandatory_files(filename, rpm_ql_insights_client):
 
 @pytest.mark.usefixtures("register_subman")
 def test_client_logfiles_mask(insights_client):
-    """Verify that files in /var/log/insights-client have the right mode: 0600
-
-    Ref: https://bugzilla.redhat.com/show_bug.cgi?id=1955724
+    """
+    :id: 8f24500c-d0ff-4ab1-a2ae-3b99cbf68e36
+    :title: Verify Client logfiles Mask
```
Comment (ptoscano) [inline integration-tests/test_client.py:89]: this is not about a file/permission mask (like umask), but rather about the permissions themselves

**Review by ptoscano — COMMENTED** [review]
There is a bit of inconsistency in the various steps & results that talk about registering with `insights-client`:

Steps:

>         2. Register insights-client
>         1. Register the insights-client to generate log files
>         2. Register insights-client and perform pazload operations
>         1. Register the insights-client

Results:

>         2. The insights-client registers successfully
>         1. Insights-client is registered and log files are generated
>         2. The insights-client registers successfully and logs are populated
>         1. Insights-client is registrated
>         1. Insights-client is registered

Ignoring the typos and the numbers, what I personally suggest is:
- keep the step & result to only mention the registration & its successfull result
- move other bits to own steps & results
- use a single wording, for example:
  - [step] Register insights-client
  - [result] Insights-client is registered successfully

This approach applies also to the testimony docstrings added in other PRs.

---------------

I see `testimony` validation failures for all the tests like the following:
```
test_client_files_permission:19
-------------------------------

* Tokens with invalid values:
  Tier: Tier 0
      type: choice
      case sensitive: False
      choices: ['tier 1', 'tier 2', 'tier 3']
```
I'm using the `testimony.yml` as currently available in #282, which has:
```yaml
Tier:
  casesensitive: false
  required: true
  type: choice
  choices:
    - Tier 1
    - Tier 2
    - Tier 3
```
Hm, I wonder whether this seems like a bug in `testimony`...

File: integration-tests/test_client.py [hunk 4]
```diff
@@ -1,3 +1,12 @@
+"""Global Values
```
[the remark this answers is not in the document]
Comment (zpetrace) [reply integration-tests/test_client.py:1]: It's not, it convenieces me but we can omit that.

File: integration-tests/test_client.py (hunk 1 above)
Comment (zpetrace) [reply integration-tests/test_client.py:142]: of insights-client.log and insights-client-payload.log

**Comment by zpetrace** [comment]
I tried to get rid of the inconsistencies in wordings and also correct the typos etc. However I am not sure I understood the
> move other bits to own steps & results
part. If you can explain some more, please.

File: integration-tests/test_client.py [hunk 5]
```diff
@@ -131,8 +189,20 @@ def test_verify_logrotate_feature(insights_client):
...
+    :description: Verify that the file /var/lib/insights/insights-client.json exists
+    :tier: Tier 1
+    :steps:
+        1. Register insights-client
+        2. Delete /var/lib/insights/insights-client.json if it exists
+        3. run the --check-results command
+        4. Verify the existence of /var/lib/insights/insights-client.json
+    :expectedresults:
+        1. Insights-client is registered
+        2. The file /var/lib/insights/insights-client.json does not exists
```
Comment (m-horky) [inline integration-tests/test_client.py:203]: `s/exists/exist`

File: integration-tests/test_client.py [hunk 6]
```diff
@@ -131,8 +189,20 @@ def test_verify_logrotate_feature(insights_client):
...
-    This test verifies that /var/lib/insights/insights-details.json exists
-    when --check-results is called
+    :id: 2ccc8e00-0e76-47fd-bdb2-27998c0094ab
+    :title: Verify insights-client details file exists
+    :description: Verify that the file /var/lib/insights/insights-client.json exists
+    :tier: Tier 1
+    :steps:
+        1. Register insights-client
+        2. Delete /var/lib/insights/insights-client.json if it exists
+        3. run the --check-results command
```
Comment (m-horky) [inline integration-tests/test_client.py:199]: Should that start with capital R?

**Comment by zpetrace** [comment]
I have changed the capitalization to the same format for every test as it was confusing. Hope that's better:)

**Review by ptoscano — APPROVED** [review]

**Pull request merged**